models/issues/pull.go

```diff
@@ -430,6 +430,24 @@ func (pr *PullRequest) GetGitHeadBranchRefName() string {
 	return fmt.Sprintf("%s%s", git.BranchPrefix, pr.HeadBranch)
 }
 
+// GetReviewCommentsCount returns the number of review comments made on the diff of a PR review (not including comments on commits or issues in a PR)
+func (pr *PullRequest) GetReviewCommentsCount(ctx context.Context) int {
+	opts := FindCommentsOptions{
+		Type:    CommentTypeReview,
+		IssueID: pr.IssueID,
+	}
+	conds := opts.ToConds()
+	if pr.ID == 0 {
+		conds = conds.And(builder.Eq{"invalidated": false})
+	}
```

Comment:
It looks counter-intuitive. Why `pr.ID == 0` means `invalidated==false`?

Reply:
Honestly, I can't answer that question as I basically copied this function from here https://github.com/go-gitea/gitea/blob/ea8e4baacc5c58e45e68291334c3d2c42e9d6737/models/issues/review.go#L995

It could very well be that the entire check there is redundant and not needed for the function to work properly. 
However, in order to evaluate that I need to revive my dev setup for this project. I'm a bit busy right now so it might take some days.

Reply:
@6543 the code `GetCodeCommentsCount` was written in API: Add pull review endpoints (#11224)

Do you have ideas about why checking `pr.ID == 0` ?

Reply:
ping @6543 ~~~

Reply:
I think it doesn't need this `if`, so let's remove it until there is a clear requirement.

Reply:
@wxiaoguang it had it's reason back then i guess but I cant find one now